### PR TITLE
seacas: require +metis and +mpi instead of +parmetis

### PR DESCRIPTION
**Pull request by ghost**
This change aligns the build condition for parmetis with the depends_on condition.
The current build condition of parmetis looks for "+parmetis" in the spec which is not added by the depends_on as that adds "^parmetis" instead.

### CODE REVIEW

**Comment by ghost** [comment]
@psakievich

**Review by psakievich — APPROVED** [review]

**Comment by psakievich** [comment]
Holding off on merging for @gsjaardema as the package maintainer to sign off.

**Review by gsjaardema — APPROVED** [review]

**Pull request merged**
